### PR TITLE
Refactor Flight Control

### PR DESCRIPTION
**Pull request by yameatmeyourdead**

### CODE REVIEW

File: src/rov/flight_controller/include/flight_controller/Thruster.hpp [hunk 1]
```diff
@@ -1,14 +1,69 @@
...
+            return -1.23 + 15.9 * throttle + 23.2 * throttle * throttle;
+        } else {
+            // negative thrust curve derived from BlueRobotics' testing
+            return 1.05 + 13.5 * throttle - 17.1 * throttle * throttle;
+        }
+    }
+    return 0;
+}
+
+inline static float throttle_from_thrust(const float thrust, bool clamping = true) {
```
Comment (Hermanoid) [inline src/rov/flight_controller/include/flight_controller/Thruster.hpp:41]: Remove this stuff?

File: src/rov/flight_controller/include/flight_controller/flight_controller.hpp [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    void toggle_PID(const std_srvs::srv::Empty_Request::SharedPtr request, std_srvs::srv::Empty_Response::SharedPtr response);
+    void setpoint_callback(const rov_interfaces::msg::ThrusterSetpoints::SharedPtr setpoints);
+    void bno_callback(const rov_interfaces::msg::BNO055Data::SharedPtr bno_data);
+    void updateSimple();
+    void updatePID();
+    void normalizethrottles(Eigen::Matrix<double,NUM_THRUSTERS,1>* throttles);
+
+    rclcpp::Subscription<rov_interfaces::msg::ThrusterSetpoints>::SharedPtr thruster_setpoint_subscription;
+    rclcpp::Subscription<rov_interfaces::msg::BNO055Data>::SharedPtr bno_data_subscription;
+    rclcpp::Publisher<rov_interfaces::msg::PWM>::SharedPtr _publisher;
```
Comment (Hermanoid) [inline src/rov/flight_controller/include/flight_controller/flight_controller.hpp:33]: Publisher of what? Name your duds! Professor Baldwin just released V20 of the coding standards document and you're going to be docked points!!

**Review by Hermanoid — CHANGES_REQUESTED** [review]
Remove old dumb code. Otherwise good.

**Comment by yameatmeyourdead** [comment]
Changes implemented, fixes applied, physical tests run. As there are 53 commits in this PR im going to squash and merge to not continue to pollute our already over polluted git tree

**Pull request merged**